### PR TITLE
[#5] Added example demonstrating how to use GenQuery2 microservices with the Python REP

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -299,3 +298,35 @@ NOTE: GenQuery2 column names are subject to change as the implementation is impr
...
+### How do I use the GenQuery2 microservices with the Python Rule Engine Plugin?
+
+```python
+def list_all_data_objects(rule_args, callback, rei):
+    # The Python Rule Engine Plugin does not allow input arguments to be modified directly.
+    #
+    # To get around this, you must retrieve the object from the "arguments" list in the
+    # dictionary returned by the microservice.
+    #
+    # Notice how the in/out parameters now act as placeholders to the microservice calls.
```
Comment (korydraughn) [inline README.md:313]: This can be better.

Comment (korydraughn) [reply README.md]: Updated.

**Comment by korydraughn** [comment]
Any thoughts on the wording?

File: README.md [hunk 2]
```diff
@@ -299,3 +298,39 @@ NOTE: GenQuery2 column names are subject to change as the implementation is impr
 | TICKET_ALLOWED_USER_TICKET_ID | R_TICKET_ALLOWED_USERS | ticket_id |
 | TICKET_ALLOWED_GROUP_NAME | R_TICKET_ALLOWED_GROUPS | group_name |
 | TICKET_ALLOWED_GROUP_TICKET_ID | R_TICKET_ALLOWED_GROUPS | ticket_id |
+
+## Questions and Answers
+
+### Can I use the GenQuery2 microservices with the Python Rule Engine Plugin (PREP)?
+
+Yes! The important thing to remember is that the GenQuery2 microservices can update input arguments. This behavior is fully supported by the iRODS Rule Language, but not the PREP. Changes to input arguments are only visible via the object returned from the microservice.
```
Comment (trel) [inline README.md:306]: I notice an exclamation point - perhaps a bit much - but okay to leave it if you feel strongly.

File: README.md [hunk 3]
```diff
@@ -299,3 +298,39 @@ NOTE: GenQuery2 column names are subject to change as the implementation is impr
...
+```python
+def list_all_data_objects(rule_args, callback, rei):
+    # Notice the first input argument. In the iRODS Rule Language, the microservice
+    # would write the context handle to it. Here, it is only a placeholder.
+    res = callback.genquery2_execute('', 'select COLL_NAME, DATA_NAME')
+
+    # We can retrieve the context handle by reading the "arguments" list like so. The
+    # "arguments" list is zero-indexed and each index maps to an argument in the call
+    # previously made.
+    ctx = res['arguments'][0]
```
Comment (trel) [inline README.md:319]: consider 'context' and 'result' in the example...   but everything else seems good and correct.

File: README.md (hunk 2 above)
Comment (korydraughn) [reply README.md:306]: I don't have any attachment to it.

Will remove.

File: README.md (hunk 3 above)
Comment (korydraughn) [reply README.md:319]: Sure thing.

File: README.md (hunk 2 above)
Comment (korydraughn) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (korydraughn) [reply README.md]: Changed those everywhere for consistency.

**Comment by korydraughn** [comment]
I think this is in a good spot. Squashing and adding the pound.

If the README needs to be tweaked more, we can do that in a separate PR.

**Review by trel — APPROVED** [review]

**Pull request merged**